File: sylee723/4주차/PGS_k진수에서_소수_개수_구하기.java
import java.util.*;

class Solution {
    public int solution(int n, int k) {
        String numStr = nToKBase(n, k);
        int answer = 0;
        
        for (String num : numStr.split("0")) {
            if (!num.equals("") && isPrime(Long.parseLong(num)))
                answer++;
        }
        
        return answer;
    }
    
    static String nToKBase(int n, int k) {
        String result = "";
        
        while (n > 0) {
            result = (n % k) + result;
            n /= k;
        }
        
        return result;
    }
    
    static boolean isPrime(long num) {
        if (num == 1)
            return false;
        
        int sqrt = (int)Math.sqrt(num);
        for (int i = 2; i <= sqrt; i++) {
            if (num % i == 0)
                return false;
        }
        
        return true;
    }
}

File: daehwa_park/4주차/PGS_k진수에서_소수_개수_구하기.py
def solution(n, k):
    answer = 0
    
    num = n_to_k(n, k)
    filtered = [p for p in num.split("0") if p]
    for p in map(int, filtered):
        if is_prime(p):
            answer += 1
    
    return answer

def n_to_k(n, k):
    num = ""
    while n > 0:
        q, r = divmod(n, k)
        num = str(r) + num
        n = q
    return num

def is_prime(num):
    if num == 1:
        return False
    for i in range(2, int(num ** 0.5) + 2):
        if num != i and num % i == 0:
            return False
    return True